### PR TITLE
Adding support for server side encryption of spark jar.

### PR DESCRIPTION
**Pull request by alvinhenrick**
Hello, Your plugin is very helpful . I have added the support for spark jar file with server side encryption upload support . Could you please merge and release it to maven central ?

Let me know if you need anything from me.

Warm Regards,
Alvin.

### CODE REVIEW

File: src/main/scala/EmrSparkPlugin.scala [hunk 1]
```diff
@@ -307,7 +307,14 @@ object EmrSparkPlugin extends AutoPlugin {
...
     val s3Jar = new S3Url(sparkS3JarFolder.value) / jar.getName
     log.info(s"Putting ${jar.getPath} to ${s3Jar.toString}")
-    sparkS3ClientBuilder.value.build().putObject(s3Jar.bucket, s3Jar.key, jar)
+    if (!sparkS3SSEEnabled.value) {
+      sparkS3ClientBuilder.value.build().putObject(s3Jar.bucket, s3Jar.key, jar)
+    } else {
+      val objectMetadata = new ObjectMetadata
+      objectMetadata.setSSEAlgorithm(ObjectMetadata.AES_256_SERVER_SIDE_ENCRYPTION)
+      val putRequest = new PutObjectRequest(s3Jar.bucket, s3Jar.key, jar)
+      sparkS3ClientBuilder.value.build().putObject(putRequest)
```
Comment (pishen) [inline src/main/scala/EmrSparkPlugin.scala]: Should this be
```scala
sparkS3ClientBuilder.value.build().putObject(putRequest.withMetadata(objectMetadata))
```
?

File: src/main/scala/EmrSparkPlugin.scala [hunk 2]
```diff
@@ -66,6 +63,8 @@ object EmrSparkPlugin extends AutoPlugin {
       settingKey[JobFlowInstancesConfig]("default JobFlowInstancesConfig")
     val sparkRunJobFlowRequest =
       settingKey[RunJobFlowRequest]("default RunJobFlowRequest")
+    val sparkS3SSEEnabled =
+      settingKey[Boolean]("Upload Spark Jar with server side encryption")
```
Comment (pishen) [inline src/main/scala/EmrSparkPlugin.scala]: For the flexibility of customizing `PutObjectRequest`, how if we provide a function like
```scala
val sparkS3PutObjectDecorator = settingKey[PutObjectRequest => PutObjectRequest]("...")
```
and let user apply the encryption by themselves in `build.sbt` like below?
```scala
import com.amazonaws.services.s3.model.ObjectMetadata
sparkS3PutObjectDecorator := { req =>
  val metadata = new ObjectMetadata()
  metadata.setSSEAlgorithm(ObjectMetadata.AES_256_SERVER_SIDE_ENCRYPTION)
  req.withMetadata(metadata)
}
```

File: src/main/scala/EmrSparkPlugin.scala (hunk 1 above)
Comment (alvinhenrick) [reply src/main/scala/EmrSparkPlugin.scala]: You are correct somehow I missed that . Good Catch!. Will make the change and commit. Thnx

File: src/main/scala/EmrSparkPlugin.scala (hunk 2 above)
Comment (alvinhenrick) [reply src/main/scala/EmrSparkPlugin.scala]: awesome suggestion.

**Comment by alvinhenrick** [comment]
implemented the changes you suggested.By providing the function to user is awesome way to do it.Let me know if you think something is not right. Thanks for you help.

**Comment by pishen** [comment]
LGTM. Thanks for the contribution.

**Comment by pishen** [comment]
@alvinhenrick The email address in your commits is `alvin@olemedia.com`, which is not connected with your Github account. You may try to add the email to your Github account to let your name shown on the contributors list. :P

**Comment by alvinhenrick** [comment]
Sent another pull request with updated email since I no longer have access to alvin@olemedia.com email address. 
Apologize for inconvenience. Thanks

**Pull request merged**